tracing-appender/src/rolling.rs

```diff
@@ -558,17 +562,100 @@ impl Inner {
                     .unwrap_or(0),
             ),
             rotation,
+            max_files,
         };
         Ok((inner, writer))
     }
 
+    fn prune_old_logs(&self, max_files: usize) {
+        let format;
+        match self.rotation {
+            Rotation::MINUTELY => {
+                format = format_description::parse("[year]-[month]-[day]-[hour]-[minute]")
+                    .expect("Unable to create a formatter; this is a bug in tracing-appender");
+            }
+            Rotation::HOURLY => {
+                format = format_description::parse("[year]-[month]-[day]-[hour]")
+                    .expect("Unable to create a formatter; this is a bug in tracing-appender");
+            }
+            Rotation::DAILY => {
+                format = format_description::parse("[year]-[month]-[day]")
+                    .expect("Unable to create a formatter; this is a bug in tracing-appender");
+            }
+            Rotation::NEVER => {
+                unreachable!("maximum log files feature is not available for `Rotation::NEVER`!")
+            }
+        }
+
+        let files = fs::read_dir(&self.log_directory).map(|dir| {
+            dir.filter_map(|entry| {
+                let entry = entry.ok()?;
+
```

Comment:
it might also be worth checking that that the entry's metadata indicates that it's a file using [`Metadata::is_file`](https://doc.rust-lang.org/stable/std/fs/struct.Metadata.html#method.is_file). the appender will never create directories or symlinks, so if the log dir happens to contain a symlink or directory whose name matches our log file format (unlikely, but certainly possible), we shouldn't delete it.

Reply:
your PR fixes also this, so resolving this convo :)